### DISCUSSION TITLE
Curried function

### DISCUSSION THREAD

**Discussion by resynth1943** [denoland/deno#8780] [Ideas]
In F# (and this is one of my favourite features), `printf` is automatically curried, so you can do things like this (JS example):

```js
[1, 2, 3].map(printf("My number is %d")).map(console.log);
```

This may not be feasible to implement; it would most likely require quite a lot of refactoring, and LSP / compiler support (which probably isn't happening).

Just an idea, though.

**Comment by kitsonk** [comment]
Deno is a runtime for TypeScript and JavaScript. It does not create it's own language structures. There are a few proposals for JavaScript, like a competing pipeline operator, that would bring some of these features. Deno adopts these features once incorporated in v8 and TypeScript rather quickly.

**Comment by willin** [comment]
i wonder when pipeline operator can be supported

**Reply by kitsonk** [reply]
When it is progresses to Stage 3 and is implemented by TypeScript, V8 and swc.
